src/main/scala/fi/aalto/cs/apluscourses/intellij/Repl.scala

```diff
@@ -0,0 +1,20 @@
+package fi.aalto.cs.apluscourses.intellij
+
+import com.intellij.execution.ui.ConsoleViewContentType
+import com.intellij.openapi.module.Module
+import fi.aalto.cs.apluscourses.intellij.utils.ModuleUtils.{getReplInitialCommandsForModule, getUpdatedText}
+import org.jetbrains.plugins.scala.console.ScalaLanguageConsole
+
+class Repl(module: Module) extends ScalaLanguageConsole(module: Module) {
+
+  override def print(text: String, contentType: ConsoleViewContentType): Unit = {
+    var updatedText = text
+
+    if (text.contains("Type in expressions")) {
```

Comment:
This is quite a risky way to check that.

Would it be possible to at least make sure that the initial commands are not used to override actual things more than once per REPL?

Reply:
thanks, good point! I'll extend the check to verify the content is of a "WelcomeContentType" to make it sure :bowing_man:

Reply:
made some improvements (a flag and a better `String` to check against), but it will remain hackish, as we're intercepting here...